File: flutter/lib/widgets/token_management_card.dart
import 'package:demo_krdpass_auth/theme.dart';
import 'package:flutter/material.dart';

/// Verify / refresh / revoke. The transient result line lives in MainScreen: "Tokens
/// revoked, signed out" has to outlive this card, which disappears the moment the
/// revoke succeeds.
class TokenManagementCard extends StatelessWidget {
  const TokenManagementCard({
    required this.busy,
    required this.onVerifyToken,
    required this.onRefreshToken,
    required this.onRevokeToken,
    super.key,
  });

  final bool busy;
  final VoidCallback onVerifyToken;
  final VoidCallback onRefreshToken;
  final VoidCallback onRevokeToken;

  @override
  Widget build(BuildContext context) {
    final colors = context.krdpassColors;
    return Card(
      elevation: 0,
      color: Theme.of(context).cardTheme.color,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: BorderSide(color: colors.line.withValues(alpha: 0.5)),
      ),
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          children: [
            Row(
              children: [
                Icon(
                  Icons.settings_rounded,
                  size: 20,
                  color: Theme.of(context).colorScheme.primary,
                ),
                const SizedBox(width: 12),
                Text(
                  'Token Management',
                  style: TextStyle(
                    fontSize: 14,
                    fontWeight: FontWeight.bold,
                    color: Theme.of(context).colorScheme.onSurface,
                  ),
                ),
              ],
            ),
            const SizedBox(height: 16),
            Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                _ActionButton(
                  label: 'Verify Token Signature',
                  color: Theme.of(context).colorScheme.primary,
                  onPressed: busy ? null : onVerifyToken,
                ),
                const SizedBox(height: 8),
                _ActionButton(
                  label: 'Refresh Access Token',
                  color: colors.success,
                  onPressed: busy ? null : onRefreshToken,
                ),
                const SizedBox(height: 8),
                _ActionButton(
                  label: 'Revoke Token (Log Out)',
                  color: Theme.of(context).colorScheme.error,
                  onPressed: busy ? null : onRevokeToken,
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

class _ActionButton extends StatelessWidget {
  const _ActionButton({
    required this.label,
    required this.color,
    required this.onPressed,
  });

  final String label;
  final Color color;
  final VoidCallback? onPressed;

  @override
  Widget build(BuildContext context) {
    return FilledButton(
      onPressed: onPressed,
      style: FilledButton.styleFrom(
        backgroundColor: color.withValues(alpha: 0.1),
        foregroundColor: color,
        fixedSize: const Size(double.infinity, 44),
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
      ),
      child: Text(label),
    );
  }
}

File: android/app/src/main/java/krd/pass/auth/demo/ui/TokenManagementCard.kt
package krd.pass.auth.demo.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import krd.pass.auth.demo.ui.theme.KrdpassSuccess

/** Verify / refresh / revoke. The transient result line lives in MainScreen: "Tokens revoked,
 * signed out" has to outlive this card, which disappears the moment the revoke succeeds. */
@Composable
fun TokenManagementCard(
    loading: Boolean = false,
    onVerifyToken: () -> Unit,
    onRefreshToken: () -> Unit,
    onRevokeToken: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        border = BorderStroke(1.dp, MaterialTheme.colorScheme.outlineVariant.copy(alpha = 0.3f)),
        elevation = CardDefaults.cardElevation(0.dp)
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Default.Settings, contentDescription = null, tint = MaterialTheme.colorScheme.primary, modifier = Modifier.size(20.dp))
                Spacer(Modifier.width(12.dp))
                Text("Token Management", style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold), color = MaterialTheme.colorScheme.onSurface)
            }

            Spacer(Modifier.height(16.dp))

            Column(modifier = Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Button(
                    onClick = onVerifyToken,
                    enabled = !loading,
                    modifier = Modifier.fillMaxWidth().height(44.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.1f), contentColor = MaterialTheme.colorScheme.primary),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Text("Verify Token Signature", style = MaterialTheme.typography.labelLarge)
                }

                Button(
                    onClick = onRefreshToken,
                    enabled = !loading,
                    modifier = Modifier.fillMaxWidth().height(44.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = KrdpassSuccess.copy(alpha = 0.1f), contentColor = KrdpassSuccess),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Text("Refresh Access Token", style = MaterialTheme.typography.labelLarge)
                }

                Button(
                    onClick = onRevokeToken,
                    enabled = !loading,
                    modifier = Modifier.fillMaxWidth().height(44.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error.copy(alpha = 0.1f), contentColor = MaterialTheme.colorScheme.error),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Text("Revoke Token (Log Out)", style = MaterialTheme.typography.labelLarge)
                }
            }
        }
    }
}
